elyra/validation/validate.py

```diff
@@ -0,0 +1,424 @@
+            supported_ops.append('execute-python-node')
+        if version.parse(elyra_version) >= version.parse("2.2.0"):
+            supported_ops.append('execute-r-node')
+        supported_runtimes = ['kfp', 'airflow', 'null']  # need to rethink this null string
+        support_matrix = {}
+        for runtime in supported_runtimes:
+            support_matrix[runtime] = supported_ops
+
+        # Checks pipeline node types are compatible with the runtime selected
+        pipeline_runtime = pipeline_json['pipelines'][0]['app_data']['runtime']
+        node_list = pipeline_json['pipelines'][0]['nodes']
+
+        for node in node_list:
+            if node['op'] not in support_matrix.get(pipeline_runtime, []):
+                response.add_message(severity='1',
+                                     message_type="invalidNodeType",
+                                     message="Unsupported node type found in this pipeline",
+                                     data={"nodeID": node['id'],
+                                           "pipelineID": pipeline_id})
+
+    def validate_node_properties(self, root_dir: str, pipeline: Dict, response):
+        """
+        Validates each of the node's structure for required fields/properties as well as
+        their values
+        :param root_dir: the absolute base path of the current elyra workspace
+        :param pipeline: the pipeline definition to be validated
+        :param response: ValidationResponse containing the issue list to be updated
+        """
+        pipeline_json = json.loads(json.dumps(pipeline))
+        node_list = pipeline_json['pipelines'][0]['nodes']
+
+        # Validate the values for each of the properties in the node"""
+        # TODO: Validate against more specific node properties in component registry
+
+        for node in node_list:
+            node_data = node['app_data']
+
+            # Validate the node structure
+            node_field_strings = ['filename', 'runtime_image', 'include_subdirectories', 'cpu', 'gpu', 'memory']
+
+            for field_string in node_field_strings:
+                if field_string not in node_data:
+                    response.add_message(severity='1',
+                                         message_type="invalidNodeProperty",
+                                         message="Node is missing field",
+                                         data={"nodeID": node['id'],
+                                               "propertyName": field_string})
+                elif not isinstance(node_data[field_string], str):
+                    response.add_message(severity='1',
+                                         message_type="invalidNodeProperty",
+                                         message="Node field is incorrect type",
+                                         data={"nodeID": node['id'],
+                                               "propertyName": field_string})
```

Comment:
Same comment, but have the method relative to data-type (validate_field_type).  The two could then be wrapped in another that adds type to the signature (e.g., 'validate_field()`)

These could then be applied below.